CHANGELOG.md

```diff
@@ -1,6 +1,201 @@
+     // position.
+     rpc GetFeature(Point) returns (Feature) {}
+   
+     // A server-to-client streaming RPC.
+     //
+     // Obtains the Features available within the given Rectangle.  Results are
+     // streamed rather than returned at once (e.g. in a response message with a
+     // repeated field), as the rectangle may cover a large area and contain a
+     // huge number of features.
+     rpc ListFeatures(Rectangle) returns (stream Feature) {}
+   
+     // A client-to-server streaming RPC.
+     //
+     // Accepts a stream of Points on a route being traversed, returning a
+     // RouteSummary when traversal is completed.
+     rpc RecordRoute(stream Point) returns (RouteSummary) {}
+   
+     // A Bidirectional streaming RPC.
+     //
+     // Accepts a stream of RouteNotes sent while a route is being traversed,
+     // while receiving other RouteNotes (e.g. from other users).
+     rpc RouteChat(stream RouteNote) returns (stream RouteNote) {}
+   }    
+   ```
+   
+   The generated code will look like the following (message protos, referenced by the schema, are
+   omitted):
+   
+   ```kotlin
+   interface RouteGuide : Service {
+     @WireRpc(
+         path = "/routeguide.RouteGuide/GetFeature",
+         requestAdapter = "routeguide.Point#ADAPTER",
+         responseAdapter = "routeguide.Feature#ADAPTER"
+     )
+     suspend fun GetFeature(request: Point): Feature
+   
+     @WireRpc(
+         path = "/routeguide.RouteGuide/ListFeatures",
+         requestAdapter = "routeguide.Rectangle#ADAPTER",
+         responseAdapter = "routeguide.Feature#ADAPTER"
+     )
+     suspend fun ListFeatures(request: Rectangle): ReceiveChannel<Feature>
+   
+     @WireRpc(
+         path = "/routeguide.RouteGuide/RecordRoute",
+         requestAdapter = "routeguide.Point#ADAPTER",
+         responseAdapter = "routeguide.RouteSummary#ADAPTER"
+     )
+     suspend fun RecordRoute(): Pair<SendChannel<Point>, Deferred<RouteSummary>>
+   
+     @WireRpc(
+         path = "/routeguide.RouteGuide/RouteChat",
+         requestAdapter = "routeguide.RouteNote#ADAPTER",
+         responseAdapter = "routeguide.RouteNote#ADAPTER"
+     )
+     suspend fun RouteChat(): Pair<SendChannel<RouteNote>, ReceiveChannel<RouteNote>>
+   }
+   ```
+   
+   All four gRPC modes are supported: the generated code uses suspendable functions to implement
+   non-blocking asynchronous execution. In streaming modes, `ReceiveChannel` and `SendChannel` are
+   use to listen to asynchronous data in a non-blocking fashion.
+   
+   This feature works out-of-the-box in Wire 3 compiler as long as the input file contains a gRPC
+   schema.
+ 
+ * New: Gradle plugin
+ 
+   Historically Wire only provided command line compiler API and a Maven plugin. As Gradle gained
+   popularity and is a de-facto standard in Android app development, a Wire Gradle plugin became a
+   necessity.
+   
+   Here's an example of Gradle configuration:
```

Comment:
This is one where the full example is definitely worth keeping!

Reply:
@jrodbx I'd like to mention proto dependencies, is there an example that includes them?